### PR TITLE
fix strong mode errors, turn on strong mode analysis

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: lib/src/collection/multimap.dart [hunk 1]
```diff
@@ -106,8 +108,8 @@ abstract class _BaseMultimap<K, V, C extends Iterable<V>>
   /// to the identity function.
   _BaseMultimap.fromIterable(Iterable iterable,
       {K key(element), V value(element)}) {
-    key ??= _id;
-    value ??= _id;
+    key ??= _id as _IdGetter<K>;
```
Comment (jakemac53) [inline lib/src/collection/multimap.dart]: pushing a better solution for this actually

**Comment by jakemac53** [comment]
cc @davidmorgan

**Comment by cbracken** [comment]
lgtm

Looks like a diff in `dartfmt` output between stable and dev trees. There's probably no winning this one on this PR, but I'll land a patch that limits this check to the stable SDK.

**Comment by cbracken** [comment]
Alright landed my fix and re-kicked off Travis on this PR. You're good to go. Thanks!

**Comment by jakemac53** [comment]
I don't have write access to this repo so please merge at will :)

**Pull request merged**